alvr/server/cpp/platform/linux/EncodePipelineSW.cpp

```diff
@@ -41,70 +45,37 @@ alvr::EncodePipelineSW::EncodePipelineSW(Renderer *render, uint32_t width, uint3
     throw std::runtime_error("HEVC is not supported by SW encoder");
   }
 
-  auto codec_id = ALVR_CODEC(settings.m_codec);
-  const char * encoder_name = encoder(codec_id);
-  const AVCodec *codec = avcodec_find_encoder_by_name(encoder_name);
-  if (codec == nullptr)
-  {
-    throw std::runtime_error(std::string("Failed to find encoder ") + encoder_name);
+  x264_param_default_preset(&param, "ultrafast", "zerolatency");
+  x264_param_apply_profile(&param, "high");
+
+  param.pf_log = x264_log;
+  param.i_log_level = X264_LOG_INFO;
+
+  param.b_aud = 0;
+  param.b_cabac = settings.m_entropyCoding == ALVR_CABAC;
+  param.b_sliced_threads = true;
+  param.i_threads = settings.m_swThreadCount;
+  param.i_width = width;
+  param.i_height = height;
+  param.i_fps_num = 60 * 1'000'000;
+  param.i_fps_den = 1'000'000;
+  param.rc.i_rc_method = X264_RC_ABR;
+  param.rc.i_bitrate = 30'000;
+  param.rc.i_vbv_buffer_size = param.rc.i_bitrate / 60 * 1.1;
+  param.rc.i_vbv_max_bitrate = param.rc.i_bitrate;
+  param.rc.f_vbv_buffer_init = 0.75;
+
+  enc = x264_encoder_open(&param);
```

Comment:
It could be nice to check that ` x264_encoder_maximum_delayed_frames` is 0, as some configuration options may introduce delayed frames.

Reply:
If that happens this should catch it:

https://github.com/alvr-org/ALVR/blob/master/alvr/server/cpp/platform/linux/CEncoder.cpp#L248